components/camel-restlet/src/test/java/org/apache/camel/component/restlet/RestletRouteBuilderTest.java

```diff
@@ -75,18 +78,34 @@ public void process(Exchange exchange) throws Exception {
                             + exchange.getIn().getHeader("x"));
                     }
                 });
+                
+                // Restlet consumer to handler FORM POST method
+                from("restlet:http://localhost:" + portNum + "/login?restletMethod=post").process(new Processor() {
+                    public void process(Exchange exchange) throws Exception {
+                        exchange.getOut().setBody(
+                            "received user: "
+                            + exchange.getIn().getHeader("user")
+                            + "password: "
+                            + exchange.getIn().getHeader("passwd"));
+                    }
+                });
             }
         };
     }
 
     @Test
     public void testProducer() throws IOException {
-        String response = template.requestBody("direct:start", "<order foo='1'/>", String.class);
+        Map<String, Object> headers = new HashMap<String, Object>();
+        headers.put(Exchange.CONTENT_TYPE, MediaType.APPLICATION_XML);
```

Comment:
I had to add the new code otherwise this unit test fails when setting the correct content type.  (All request default to APPLICATION_WWW_FORM if content-type is not set).  

Without the new code, the headers (id, 89531) do not get passed through from the request.  
You can test this in the master branch by just changing this test to use the correct content-type, it will fail.